Source/Core/DolphinQt/Config/LogWidget.cpp

```diff
@@ -60,36 +73,50 @@ LogWidget::~LogWidget()
 
 void LogWidget::UpdateLog()
 {
-  std::lock_guard<std::mutex> lock(m_log_mutex);
-
-  if (m_log_queue.empty())
-    return;
-
-  auto* vscroll = m_log_text->verticalScrollBar();
-  auto* hscroll = m_log_text->horizontalScrollBar();
+  std::vector<std::pair<std::string, LogTypes::LOG_LEVELS>> elements_to_push;
+  {
+    std::lock_guard lock(m_log_mutex);
+    if (m_log_queue.empty())
+      return;
 
-  // If the vertical scrollbar is within 50 units of the maximum value, count it as being at the
-  // bottom
-  bool vscroll_bottom = vscroll->maximum() - vscroll->value() < 50;
+    elements_to_push.reserve(MAX_LOG_LINES);
```

Comment:
Would this be more practical as `m_log_queue.size()` instead, since there can be at most `MAX_LOG_LINES` in there, with a good chance that there is less (for non-debug builds with non-spammy listeners)?

Reply:
`m_log_queue` size can be `MAXIMUM_BLOCK_COUNT` at most. `MAX_LOG_LINES` is the maximum amount of log lines to be outputted to the widget in one iteration.

That said, `std::min(MAX_LOG_LINES, m_log_queue.size())` is indeed a good idea here :)

Reply:
I should also probably rename those constants as this proves they're confusingly named.

Reply:
Oops, mixing up constants, but yeah; you caught my drift 👍